tfx_bsl/tfxio/parquet_tfxio.py

```diff
@@ -0,0 +1,133 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TFXIO implementation for Parquet."""
+
+import copy
+from typing import Optional, List, Text, Any
+
+import apache_beam as beam
+import pyarrow as pa
+import tensorflow as tf
+from tensorflow_metadata.proto.v0 import schema_pb2
+
+from tfx_bsl.coders import csv_decoder
+
+from tfx_bsl.tfxio import dataset_options, tensor_adapter, tensor_representation_util
+from tfx_bsl.tfxio.tfxio import TFXIO
+
+
+class ParquetTFXIO(TFXIO):
+  """TFXIO implementation for Parquet."""
+
+  def __init__(self,
+               file_pattern: Text,
+               column_names: List[Text],
+               min_bundle_size: int = 0,
+               schema: Optional[schema_pb2.Schema] = None,
+               validate: Optional[bool] = True):
+    """Initializes a Parquet TFXIO.
+
+    Args:
+      file_pattern: A file glob pattern to read parquet files from.
+      column_names: List of column names to read from the parquet files.
+      min_bundle_size: the minimum size in bytes, to be considered when
+        splitting the parquet input into bundles.
+      schema: An optional TFMD Schema describing the dataset. If schema is
+        provided, it will determine the data type of the csv columns. Otherwise,
+        the each column's data type will be inferred by the csv decoder. The
+        schema should contain exactly the same features as column_names.
+      validate: Boolean flag to verify that the files exist during the pipeline
+        creation time.
+    """
+    self._file_pattern = file_pattern
+    self._column_names = column_names
+    self._min_bundle_size = min_bundle_size
+    self._validate = validate
+    self._schema = schema
+
+  def BeamSource(self, batch_size: Optional[int] = None) -> beam.PTransform:
+
+    @beam.typehints.with_input_types(Any)
+    @beam.typehints.with_output_types(pa.RecordBatch)
+    def _PTransformFn(pcoll_or_pipeline: Any):
+      """Converts raw records to RecordBatches."""
+      return (pcoll_or_pipeline |
+              "ParquetBeamSource" >> beam.io.ReadFromParquetBatched(
+            file_pattern=self._file_pattern,
+            min_bundle_size=self._min_bundle_size,
+            validate=self._validate,
+            columns=self._column_names) | "ToRecordBatch" >>
+              beam.FlatMap(self._TableToRecordBatch, batch_size))
+
+    return beam.ptransform_fn(_PTransformFn)()
+
+  def RecordBatches(self, options: dataset_options.RecordBatchesOptions):
+    raise NotImplementedError
+
+  def TensorFlowDataset(
+      self, options: dataset_options.TensorFlowDatasetOptions
+  ) -> tf.data.Dataset:
+    raise NotImplementedError
+
+  def _TableToRecordBatch(
+      self,
+      table: pa.Table,
+      batch_size: Optional[int] = None) -> List[pa.RecordBatch]:
+    return table.to_batches(self, max_chunksize=batch_size)
```

Comment:
this line here throws the following exception when running tests:
```
Traceback (most recent call last):
  File "apache_beam/runners/common.py", line 1198, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 718, in apache_beam.runners.common.PerWindowInvoker.invoke_process
  File "apache_beam/runners/common.py", line 843, in apache_beam.runners.common.PerWindowInvoker._invoke_process_per_window
  File "/Users/martinbomio/Spotify/Personal/tfx-bsl/tfx_bsl/tfxio/parquet_tfxio.py", line 87, in _TableToRecordBatch
    return table.to_batches(self, max_chunksize=batch_size)
  File "pyarrow/table.pxi", line 1701, in pyarrow.lib.Table.to_batches
TypeError: to_batches() got multiple values for keyword argument 'max_chunksize'
```

Reply:
I don't think `self` should be passed to `to_batches`

Reply:
🤦 of course